### ISSUE TITLE
authUI.authViewController() for email does not reset after login

### ISSUE THREAD

**Issue by yongrim** [firebase/FirebaseUI-iOS#623]
Steps to repro:

1. Login or signup using email
2. Logout
3. Login or signup again to see that the last view of the previous login/signup is displayed instead of the root view of the authUI.authViewController()

Recently it seems that isSignInWithEmailHidden flag was removed and instead now we are required to use FUIEmailAuth as one of the providers if we want to use email auth.

Since this change it looks like for the last view in the navigation stack of the email login/signup flow stays as the view presented.

This behavior is different from before and from other auth providers.

This is a problem because when the user logs out and the authUI.authViewController() is presented again the last view of the email login/signup is presented again instead of the root view of the authUI.authViewController().

**Comment by xuaninbox** [comment]
I remember I have this issue before. for me its because I **performSegue** to the next view controller in the 

`authUI(_: FUIAuth, didSignInWith authDataResult: AuthDataResult?, error: Error?) `

that caused a timing issue with

`Auth.auth().addStateDidChangeListener`

So, to solve it, **performSegue** in `Auth.auth().addStateDidChangeListener` and NOT `authUI(_: FUIAuth, didSignInWith authDataResult: AuthDataResult?, error: Error?) `

**Comment by yongrim** [comment]
> I remember I had this issue before. for me its because I **performSegue** to the next view controller in the
> 
> `authUI(_: FUIAuth, didSignInWith authDataResult: AuthDataResult?, error: Error?) `
> 
> that caused a timing issue with
> 
> `Auth.auth().addStateDidChangeListener`
> 
> So to solve it, **performSegue** in `Auth.auth().addStateDidChangeListener` and NOT `authUI(_: FUIAuth, didSignInWith authDataResult: AuthDataResult?, error: Error?) `

Thanks for the response.

I don't believe that is the reason, I simply call presentViewController() and present the authViewController.

Also as I stated in the bug, this behavior has changed from what was there before. In our app we use both FB login and Email, and FB login continues to work correctly as before. Only Email auth is affected.

**Comment by Goule** [comment]
Same problem here :)

**Comment by codemodouk** [comment]
Same here too.

**Comment by Yue-Wang-Google** [comment]
We cannot reproduce the issue with our sample Demo app ---

We tested two cases:

Case 1:

Only select "Email" with the "link" switched off in IDP.  Steps:
1. Click "Sign in" and Email Sign in view is appeared. 
2 We successfully login. And Tap "Logout"
3 Click "Sign in" again and Email Sign in view is appeared again. 


Case 2:

Select "Email" with the "link" switched off in IDP.  In addition, select one or more IDPs such as Facebook and Microsoft. Steps:
1. Click "Sign in" and Auth picker view is displayed.
2. Tap "Sign in with Email" and the email Sign in view is appeared. 
2 We successfully login. And Tap "Logout"
3 Tap "Sign in" again and the Auth picker view is displayed again. Tap "Sign in with Email" and th Email Sign in view is appeared again. 

We believe the two cases match with our spec intention well. 


You are right we removed `isSignInWithEmailHidden` flag and now we requires developers to create a FUIEmailAuth. This  makes it better align with the rest of FUIAuth classes.
